vault/auth.go

```diff
@@ -148,6 +155,44 @@ func (v *Vault) UserPassLogin() string {
 	return secret.Auth.ClientToken
 }
 
+// EC2Login - AWS EC2 auth backend
+func (v *Vault) EC2Login() string {
+	role := env.Getenv("VAULT_AUTH_AWS_ROLE")
+	mount := env.Getenv("VAULT_AUTH_AWS_MOUNT", "aws")
+
+	vars := map[string]interface{}{}
+
+	if role != "" {
+		vars["role"] = role
+	}
+
+	opts := aws.ClientOptions{}
+
+	timeout := os.Getenv("AWS_TIMEOUT")
+	if timeout != "" {
```

Comment:
There's actually no need to have this check, because `opts.Timeout`'s default is `0` anyways. There's no harm to simply setting it to `0` explicitly.

You could make things even simpler by setting `Timeout` while creating the struct (up on ln169):

```go
opts := aws.ClientOptions{
  Timeout: time.Duration(typeconv.MustAtoi(os.Getenv("AWS_TIMEOUT"))) * time.Millisecond,
}
```